crates/worker/src/worker.rs

```diff
@@ -124,30 +129,116 @@ impl RunningEngine {
                 }
             };
 
-            purgatory.add_task(&sqs_message).await;
+            // TODO: add metrics for how long it takes -
+            // adjust "visibility timeout" or receiver chan capacity
             match sqs_message {
                 SqsMessage::Compile { request } => {
-                    let result = compile(request, &db_client, &s3_client).await; // TODO:
-                    match result {
-                        Ok(()) => purgatory.update_task().await,
-                        Err(err) => {
-                            if err.recoverable() {
-                                warn!("recoverable error after compilation: {}", err);
-                                continue;
-                            } else {
-                                // delete from SQS
-                                warn!("unrecoverable error after compilation: {}", err);
-                                purgatory.update_task().await;
-                            }
-                        }
-                    }
+                    Self::process_compile_message(
+                        request,
+                        receipt_handle,
+                        &sqs_receiver,
+                        &db_client,
+                        &s3_client,
+                        &mut purgatory,
+                    )
+                    .await
+                }
+                SqsMessage::Verify { request } => {
+                    Self::process_verify_message(request, receipt_handle, &sqs_receiver).await
+                }
+            }
+        }
+    }
+
+    // TODO(future me): could return bool.
+    async fn process_compile_message(
```

Comment:
Can you try to split this function into the smaller pieces?

Reply:
I feel like it wouldn't really become any prettier than that. It's pretty self-contained as it is

Reply:
Done in latest pr